apps/hash-graph/libs/api/src/rest/data_type.rs

```diff
@@ -541,24 +531,22 @@ struct UpdateDataTypeEmbeddingsRequest {
         (status = 403, description = "Insufficient permissions to update the data type"),
         (status = 500, description = "Store error occurred"),
     ),
-    request_body = UpdateDataTypeEmbeddingsRequest,
+    request_body = UpdateDataTypeEmbeddingParams,
 )]
 #[tracing::instrument(level = "info", skip(store_pool, authorization_api_pool))]
 async fn update_data_type_embeddings<S, A>(
     AuthenticatedUserHeader(actor_id): AuthenticatedUserHeader,
     store_pool: Extension<Arc<S>>,
     authorization_api_pool: Extension<Arc<A>>,
-    body: Json<UpdateDataTypeEmbeddingsRequest>,
+    Json(body): Json<serde_json::Value>,
```

Comment:
Maybe write a comment on why this is needed. It might not be obvious to the uninitiated ones.

Reply:
I added a comment where the deserialization happens